plugins/outputs/arc/arc.go

```diff
@@ -0,0 +1,206 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package arc
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	_ "embed"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/tinylib/msgp/msgp"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	httpconfig "github.com/influxdata/telegraf/plugins/common/http"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Arc struct {
+	URL             string            `toml:"url"`
+	APIKey          config.Secret     `toml:"api_key"`
+	Database        string            `toml:"database"`
+	Headers         map[string]string `toml:"headers"`
+	ContentEncoding string            `toml:"content_encoding"`
+	httpconfig.HTTPClientConfig
+	Log telegraf.Logger `toml:"-"`
+
+	client *http.Client
+}
+
+type arcColumnarData struct {
+	Measurement string                 `msgpack:"m"`
+	Columns     map[string]interface{} `msgpack:"columns"`
+}
+
+func (*Arc) SampleConfig() string {
+	return sampleConfig
+}
+
+func (a *Arc) Init() error {
+	if a.URL == "" {
+		a.URL = "http://localhost:8000/api/v1/write/msgpack"
+	}
+
+	if a.ContentEncoding == "" {
+		a.ContentEncoding = "gzip"
+	}
+
+	if a.Timeout == 0 {
+		a.Timeout = config.Duration(5 * time.Second)
+	}
+
+	return nil
+}
+
+func (a *Arc) Connect() error {
+	ctx := context.Background()
+	client, err := a.HTTPClientConfig.CreateClient(ctx, a.Log)
```

Comment:
Please use

```suggestion
	ctx, cancel := context.WithCancel(context.Background())
	client, err := a.HTTPClientConfig.CreateClient(ctx, a.Log)
	a.cancel = cancel
```

to be able to stop potential go-routines in the http client. Call the `a.cancel` function in `Close()` before closing the idle connections.